File: javascript/8kyu/Collatz Conjecture.js
// https://www.codewars.com/kata/577a6e90d48e51c55e000217

var hotpo = function(n, acc = 0) {
    if (n <= 1) {
      return acc;
    } else {
      return hotpo(n%2==0 ? n/2 : 3*n+1, acc+1);
    }
  }

File: java/8kyu/Collatz Conjecture.java
// https://www.codewars.com/kata/577a6e90d48e51c55e000217

interface CollatzConjecture {
  static int hotpo(int n) {
    return n > 1 ? hotpo(n % 2 > 0 ? 3 * n + 1 : n / 2) + 1 : 0;
  } 
}